### ISSUE TITLE
Setting interleave_parallel_reads always results in KeyError

### ISSUE THREAD

**Issue by nobutoba** [tensorflow/datasets#2046]
**Short description**
The commits https://github.com/tensorflow/datasets/commit/e23fb248a575990cad34e260e564a37f8b181c6d, https://github.com/tensorflow/datasets/commit/c7027413b339d3a48fd309ec78ff8b2b3c93b1f4 deprecated `interleave_parallel_reads` and introduced `interleave_cycle_length` instead. However, instantiating the `ReadConfig` class with the `interleave_parallel_reads` argument always results in `KeyError`, in addition to logging the deprecation warning. An outcome of this KeyError is incompatibility of TensorFlow Dataset 3.1.0 with TensorFlow Model Garden 2.2.0.

**Environment information**
* Operating System: CentOS Linux 7
* Python version: 3.7.6
* `tensorflow-datasets`/`tfds-nightly` version: TensorFlow Datasets 3.1.0
* `tensorflow`/`tensorflow-gpu`/`tf-nightly`/`tf-nightly-gpu` version: TensorFlow 2.2.0

**Reproduction instructions**
As shown in a [README](https://github.com/tensorflow/models/blob/v2.2.0/official/vision/image_classification/README.md) of TensorFlow Model Garden 2.2.0, run the following command

```shell
python3 classifier_trainer.py \
  --mode=train_and_eval \
  --model_type=resnet \
  --dataset=imagenet \
  --data_dir=$DATA_DIR \
  --config_file=configs/examples/resnet/imagenet/gpu.yaml \
  --params_override='runtime.num_gpus=1'
```

where `DATA_DIR` is the path to the ImageNet dataset. The command throws the following `KeyError`:

```
/some/path/.venv/lib/python3.7/site-packages/tensorflow_datasets/core/utils/read_config.py in __init__(self, **kwargs)
     97         raise ValueError('Cannot set both {} and {}'.format(_OLD, _NEW))
     98       logging.warning(_WARNING_MSG)
---> 99       kwargs[_OLD] = kwargs.pop(_NEW)
    100     super(ReadConfig, self).__init__(**kwargs)

KeyError: 'interleave_cycle_length'
```

**Expected behavior**
Since the `interleave_parallel_reads` argument is still valid, it should only log a deprecation warning but must not raise an Exception as above.

**Additional context**
In a nutshell, the correct code would be

```
kwargs[_NEW] = kwargs.pop(_OLD)
```

instead of the current

```
kwargs[_OLD] = kwargs.pop(_NEW)
```

**Comment by Conchylicultor** [comment]
Thank you for reporting. Sending a fix now.
